command/TransactionReplCommand.java

```diff
@@ -125,21 +125,22 @@ public static TransactionReplCommand getCommand(LineReader reader, String prompt
             String file = tokens[1];
             command = new Source(file);
         } else {
-            String firstQueryLine = String.join(" ", tokens);
-            String query = getQuery(reader, firstQueryLine, prompt.length());
+            String query = readQuery(reader, prompt, line);
             command = new Query(query);
         }
+        if (command instanceof Query) reader.getHistory().add(command.asQuery().query().trim());
```

Comment:
We need to `trim()` before adding to history, otherwise it's not added.

Reply:
Can we replace `command instanceof Query` with `command.isQuery()`? do we have an implementation of that?